nixos/modules/services/networking/hostapd.nix

```diff
@@ -157,9 +157,8 @@ in
       { description = "hostapd wireless AP";
 
         path = [ pkgs.hostapd ];
-        wantedBy = [ "network.target" ];
-
-        after = [ "${cfg.interface}-cfg.service" "nat.service" "bind.service" "dhcpd.service" "sys-subsystem-net-devices-${cfg.interface}.device" ];
+        after = [ "sys-subsystem-net-devices-${cfg.interface}.device" ];
+        requiredBy = [ "network-link-${cfg.interface}.service" ];
```

Comment:
Maybe it should also add `BindsTo=sys-subsystem-net-devices-wlan0.device`?

Reply:
Arch does `After=network.target`, but I guess they just cannot add more specific requirements because they don't know which interface will be used.

Reply:
Yes, `BindsTo` seems appropriate. I'll switch `after` to `BindsTo` as `BindsTo` is already implies the effects of `after`.

`after=network.target` is not correct, as this potentially would hoist hostapd to the top of the dependency tree and causes the race to reappear. To get a visual idea of that, please look at the pdf attached in my PR comment. Also `after=network.target` gives a cyclic `after` chain of `hostapd->network->br0-netdev->network-link-wl4s0->hostapd`. Somebody in https://github.com/NixOS/nixpkgs/issues/16090 said they had the same bridge race with Arch. Maybe that's the reason. I'll ping this PR to them.